packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -975,7 +979,9 @@ mixin WidgetInspectorService {
   ///  * <https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#rpcs-requests-and-responses>
   ///  * [BindingBase.initServiceExtensions], which explains when service
   ///    extensions can be used.
-  void initServiceExtensions(_RegisterServiceExtensionCallback registerServiceExtensionCallback) {
+  Future<void> initServiceExtensions(_RegisterServiceExtensionCallback registerServiceExtensionCallback) async {
+    serviceInfo = await developer.Service.getInfo();
```

Comment:
should this just be 

```suggestion
    developer.Service.getInfo().then((developer.ServiceProtocolInfo info) { serviceInfo = info; };
```

It seems like we don't have to hold initializing the rest of the inspector until this call is done?

Reply:
done.